src/devices/Hx711I2c/Hx711I2c.cs

```diff
@@ -0,0 +1,349 @@
+        /// It means calibration is successfully finished.
+        /// If flashing doesn't happen and LED turns off it means calibration didn't succeed.
+        /// </summary>
+        [Command]
+        public void StartCalibration() => WriteRegisterEmpty(Hx711I2cRegister.REG_CLICK_CAL);
+
+        private int GetPeelFlag()
+        {
+            // Original code calls it Peel flag
+            // There is not much documentation on what this actually means
+            // but from testing it seems that when this register is read we
+            // get events from last time this was read:
+            // - RST was pressed
+            // - calibration is finished.
+            // Specific values are described below.
+            byte flag = ReadRegister(Hx711I2cRegister.REG_DATA_GET_PEEL_FLAG);
+            switch (flag)
+            {
+                case 1: // RST is pressed
+                case 129: // Unclear - python collapses 129
+                    return 1;
+                case 2: // calibration is finished
+                    // When CAL button is pressed the LED turns orange
+                    // then user needs to wait a bit (1-2 seconds) then place calibration weight.
+                    // Use AutomaticCalibrationWeight to set weight you use for calibration.
+                    // After that orange LED flashes 3 times and turns off.
+                    // It means calibration is finished and this register will be 2.
+                    // On unsucessful calibration (i.e. nothing was placed) there is no extra flag
+                    // and the register is still 0.
+                    return 2;
+                default:
+                    // Nothing happened
+                    return 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets weight reading. Tare should be called first.
+        /// </summary>
+        /// <returns></returns>
+        [Telemetry("Weight")]
+        public Mass GetWeight()
+        {
+            float rawReading = GetRawReading();
+
+            switch (GetPeelFlag())
+            {
+                case 1: // RST was pressed
+                    if (!IgnoreResetButton)
+                    {
+                        Tare();
+                        return Mass.FromGrams(0);
+                    }
+
+                    break;
+                case 2: // calibration is finished
+                    if (!IgnoreCalibrationButton)
+                    {
+                        CalibrationScale = GetAutomaticCalibrationScale();
+                    }
+
+                    break;
+            }
+
+            return Mass.FromGrams((rawReading - Offset) / CalibrationScale);
+        }
+
+        /// <summary>
+        /// Gets average raw reading.
+        /// </summary>
+        /// <param name="samples">Number of times sampling occurs</param>
+        /// <returns>Raw reading value</returns>
+        public float GetRawReading(int samples = 20)
```

Comment:
if you have grams, then why not using UnitsNet?

Reply:
This is not grams, it's more like ADC reading. It will be gram-like after you divide that by CalibrationScale. Gram-like because unless you pick starting point (offset/tare/zero) the weight is kinda meaningless.

It's still useful to expose it so that Offset can be set/stored appropriately if needed. If I use grams here it will be very tempting to use this value where it's not supposed to be